### DISCUSSION TITLE
Jetbrains Feature request: Toggle Copilot on/off

### DISCUSSION THREAD

**Discussion by thomasj02** [community/community#6889] [Copilot Conversations]
It would be great if there was an option to toggle copilot suggestions on and off, ideally with a hotkey that could be assigned. For example when I'm writing documentation or sketching out pseudocode, I don't want to be distracted by copilot suggestions.

**Comment by jansorg** [comment]
Thanks for reporting! 
For now there's an application setting to turn off the automatic completions, https://github.com/github/copilot-docs/blob/main/docs/jetbrains/gettingstarted.md#enabling
Would you prefer a shortcut to toggle it or rather a way to configure completions per file type?

**Reply by thomasj02** [reply]
I would definitely prefer a hotkey to toggle completions, particularly given the current state of Copilot. Copilot works great when I'm writing code that is mostly self-contained or that depends only on other code from within the same file, but once I start using classes and functions from other files it's more of a hindrance since it seems to mostly be guessing about that code.

**Reply by twigs67** [reply]
@jansorg I would prefer a way to disable suggestions on comments.

**Comment by jamesonknutson** [comment]
> It would be great if there was an option to toggle copilot suggestions on and off, ideally with a hotkey that could be assigned. For example when I'm writing documentation or sketching out pseudocode, I don't want to be distracted by copilot suggestions.

Hello there, I recognize that you are using Jetbrains, [but I wrote a VSCode extension](https://marketplace.visualstudio.com/items?itemName=disable-copilot-comment-completions.disable-copilot-comment-completions) that disables Copilot's inline suggestions while writing out comments for precisely this purpose (to avoid being distracted).

The extension is very, very simple, [and the source for it is available here](https://github.com/jamesonknutson/disable-copilot-comment-completions). You could probably re-implement this solution as a Jetbrains extension without a lot of effort, all I did was hook into the editor and toggle copilot's inline suggestions whenever the current selection included a scope with the string `comment` in it, and vice-versa; enable inline suggestions whenever the selection moved off of a comment zone.

Hacky fix, but it works for me. Maybe it can be of some use to you, as well.

**Comment by twigs67** [comment]
Is there a hero among us that will convert what @jamesonknutson has done for JetBrains? This comment suggestion on copilot is extremely annoying and has forced me to disable it.
